Makefile

```diff
@@ -6,23 +6,74 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
+.PHONY:
 generate: clean generate-protos
 
+.PHONY:
+update: clone-external-protos
+
+# Clone external repositories.
+.PHONY:
+clone-external-repos:
+	mkdir -p ExternalRepositories
+	if [ ! -d ExternalRepositories/remote-apis ]; then \
+		git -C ExternalRepositories clone --depth 1 https://github.com/bazelbuild/remote-apis; \
+	else \
+		git -C ExternalRepositories/remote-apis pull --rebase; \
+	fi
+	if [ ! -d ExternalRepositories/googleapis ]; then \
+		git -C ExternalRepositories clone --depth 1 https://github.com/googleapis/googleapis; \
+	else \
+		git -C ExternalRepositories/googleapis pull --rebase; \
+	fi
```

Comment:
Should we just use submodules, or do we want to keep this only in the Makefile?

Reply:
Good question.   I had started down the submodule route, but that does mean that everyone that references llbuild2 will clone those even when the are only used during generation.